comet-java-client/src/test/java/ml/comet/experiment/http/ConnectionUtilsTest.java

```diff
@@ -0,0 +1,123 @@
+        assertNotNull(file, "test file not found");
+
+        Request r = ConnectionUtils.createPostFileRequest(file, url, params);
+        this.validateRequest(r, url, params, HttpConstants.Methods.POST, ConnectionUtils.FORM_MIME_TYPE);
+
+        // check body parts
+        assertEquals(1, r.getBodyParts().size(), "wrong number of body parts");
+        FilePart part = (FilePart) r.getBodyParts().get(0);
+        assertEquals(ConnectionUtils.FILE, part.getName(), "wrong name");
+        assertEquals(ConnectionUtils.FORM_MIME_TYPE, part.getContentType(), "wrong content type");
+        assertEquals(file, part.getFile(), "wrong file");
+    }
+
+    @Test
+    public void testCreatePostByteArrayRequest() {
+        String url = "http://test.com" + ADD_ASSET;
+        HashMap<String, String> params = new HashMap<String, String>() {{
+            put(EXPERIMENT_KEY, "test_key");
+            put("type", ASSET_TYPE_UNKNOWN);
+            put("overwrite", Boolean.toString(true));
+        }};
+        byte[] data = "The test byte data".getBytes();
+
+        Request r = ConnectionUtils.createPostByteArrayRequest(data, url, params);
+        this.validateRequest(r, url, params, HttpConstants.Methods.POST, ConnectionUtils.FORM_MIME_TYPE);
+
+        // check body parts
+        assertEquals(1, r.getBodyParts().size(), "wrong number of body parts");
+        ByteArrayPart part = (ByteArrayPart) r.getBodyParts().get(0);
+        assertEquals(ConnectionUtils.FILE, part.getName(), "wrong name");
+        assertEquals(ConnectionUtils.FORM_MIME_TYPE, part.getContentType(), "wrong content type");
+        assertEquals(data, part.getBytes(), "wrong data array");
+    }
+
+    @Test
+    public void testCreatePostJsonRequest() {
+        String url = "http://test.com" + ADD_ASSET;
+        HtmlRest html = new HtmlRest("<html><body></body></html", "test_key",
+                false, System.currentTimeMillis());
+        String json = JsonUtils.toJson(html);
+
+        Request r = ConnectionUtils.createPostJsonRequest(json, url);
+        this.validateRequest(r, url, null, HttpConstants.Methods.POST, ConnectionUtils.JSON_MIME_TYPE);
+
+        assertEquals(json.length(), r.getBodyGenerator().createBody().getContentLength(), "wrong body");
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {200, 201, 202, 299})
+    public void testIsResponseSuccessfulTrue(int statusCode) {
+        assertTrue(ConnectionUtils.isResponseSuccessful(statusCode));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {300, 401, 404, 500})
+    public void testIsResponseSuccessfulFalse(int statusCode) {
+        assertFalse(ConnectionUtils.isResponseSuccessful(statusCode));
+    }
+
+    private void validateRequest(Request r, String url, HashMap<String, String> params,
+                                 String method, String contentType) {
+        StringBuilder buf = new StringBuilder(url);
+        if (params != null) {
```

Comment:
this is testing `org.asynchttpclient.RequestBuilder` logic. its a 3rd party. no need to test (and we are leaking http url structure implementation into this test.   in my opinion its an overkill. you can have a single test that you hard assert on the URL you expect (as a string) . same for the HTTP method. 

In addition this will be covered over the e2e test when you run vs production on github CI

Reply:
Not sure if I got your comment correctly. I'm using this method in **unit tests** of the four `ConnectionUtils` methods. These unit tests analysed by Codecov to make sure that each logical path is covered.

Three of tested methods has something similar to the:
```java
String url = "http://test.com/get";
        HashMap<String, String> params = new HashMap<String, String>() {{
            put(EXPERIMENT_KEY, "test_key");
            put("type", ASSET_TYPE_SOURCE_CODE);
            put("overwrite", Boolean.toString(false));
        }};
        Request r = ConnectionUtils.createGetRequest(url, params);

        this.validateRequest(r, url, params, HttpConstants.Methods.GET, null);
```
I was assuming that if parameters provided to the tested method it is expected that **unit test** should check somehow that exactly these parameters was processed. This is the reason for me to create utility function `validateRequest` which among other things check request parameters as well. 

In my assumption, this function is not to test logic of the `org.asynchttpclient.RequestBuilder`, but rather to test that inputs to the methods generate expected results.

I believe it is expected that **unit tests** of methods check outputs produced by method.

Reply:
As a result of our sync up meeting, we decided to leave for now.